docs/redesign-specs/redesignrequirements.md

```diff
@@ -0,0 +1,29 @@
+## Re-design Documentation
+
+#### Requirements
+
+* Remove tight coupling with external services:
+  - Hardware Isolation Layer (for network isolation and authentication).
+  - Ceph (as image data store).
+  - Software iSCSI Frameworks (TGT and IET - for diskless provisioning).
+
+
+* Remove tight coupling between CLI and BMI Components (Einstein and Database).
+
+* Enable users to replace any external components as per his requirements and resource availability (driver based model).
+  - Support for hardware iSCSI servers.
+  - Support for File/Sector based diskless provisioning.
+  - Support for different authentication services (e.g. LDAP, Kerberos, etc.).
+  - Support for different data stores (e.g. Lustre, etc).
+  - Support for different network isolators.
+
```

Comment:
I don't see the mechanisms you are using... do we have a plugin architecture?  How configure...